### PR TITLE
Supply all of the Nix-related functionality through the flake

### PR DESCRIPTION
**Pull request by exarkun**
This gathers the functionality from the various Nix source files - default.nix, tests.nix, wheel.nix, shell.nix, sources.nix, twine.nix - and exposes it all through a Nix flake.

* `nix-build` is replaced by `nix build`
* `nix-build --argstr python310 --argstr tahoe-lafs-source tahoe-lafs-master` is replaced by `nix build .#zkapauthorizer-python310-tahoe_dev`
* `nix-build tests.nix` is replaced by `nix build .#tests-pythonXX-tahoe_XX-P-C` where `P` is a hypothesis profile and `C` is "cov" or "nocov"
* `nix-build wheel.nix` is replaced by `nix build .#wheel`
* `nix shell nix/twine.nix` is replaced by `nix run .#twine`

`niv`-based workflow is replaced with `nix flake`-based workflow.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -406,42 +401,36 @@ jobs:
...
+            nix run .#mypy -- src
       - run:
           <<: *PUSH_TO_CACHIX
 
 
   # A job that merges all of the coverage data and generates a report.
   finish-coverage:
     docker:
-      - image: "nixos/nix:2.8.1"
+      - image: "nixos/nix:2.9.2"
```
Comment (hacklschorsch) [inline .circleci/config.yml:433]: Seeing this version update the third time I wonder whether there's a way to DRY this up a bit - if that makes sense?

**Review by hacklschorsch — APPROVED** [review]
Reading through the code this looks good to me. Still have to run it though.

File: .circleci/config.yml (hunk 1 above)
Comment (exarkun) [reply .circleci/config.yml:433]: The four approaches I know of:

* YAML anchors/references
* CircleCI orbs
* CircleCI custom executors
* write a well-factored Haskell program that spits out the correct yaml

**Comment by hacklschorsch** [comment]
`nix --experimental-features 'nix-command flakes' build .#zkapauthorizer-python310-tahoe_dev` works ending with  
`building '/nix/store/2wmiar17qp9j96pw0sml39lgjxzjs2m2-python3.10-zero-knowledge-access-pass-authorizer-2022.6.29.drv'...`

`nix --experimental-features 'nix-command flakes' build .#wheel` runs OK ending in  
`building '/nix/store/h1djjcgjjca0m78wkahlz8pflgpc2chm-python3.10-zero-knowledge-access-pass-authorizer-2022.6.29-wheel.drv'...`

`nix --experimental-features 'nix-command flakes' build` **errors out** with  
`error: flake 'git+file:///home/flo/Repositories/ZKAPAuthorizer' does not provide attribute 'packages.x86_64-linux.defaultPackage.x86_64-linux', 'legacyPackages.x86_64-linux.defaultPackage.x86_64-linux' or 'defaultPackage.x86_64-linux'`
- but that might be expected?

I don't know anything about flakes so far.
Stuff that is general knowledge about flakes shouldn't add up in a readme, but maybe the most common and/or useful build targets might make sense to have in the readme or contributing?

**Comment by hacklschorsch** [comment]
/reopen

I wanted to say our discussion in Open Pairing obsoletes my above comment, then clicked the wrong button.

Sorry, pressed wrong button by accident

**Pull request merged**